cmd/spire-server/cli/entry/delete.go

```diff
@@ -73,12 +118,30 @@ func (c *deleteCommand) prettyPrintDelete(env *commoncli.Env, results ...interfa
 		return cliprinter.ErrInternalCustomPrettyFunc
 	}
 
-	sts := deleteResp.Results[0].Status
-	switch sts.Code {
-	case int32(codes.OK):
-		env.Printf("Deleted entry with ID: %s\n", c.entryID)
-		return nil
-	default:
-		return fmt.Errorf("failed to delete entry: %s", sts.Message)
+	var failed, succeeded []*entryv1.BatchDeleteEntryResponse_Result
+	for _, result := range deleteResp.Results {
+		switch result.Status.Code {
+		case int32(codes.OK):
+			succeeded = append(succeeded, result)
+		default:
+			failed = append(failed, result)
+		}
 	}
+
+	for _, result := range succeeded {
+		env.Printf("Deleted entry with ID: %s\n", result.Id)
+	}
+
+	for _, result := range failed {
+		env.ErrPrintf("Failed to delete entry with ID %s (code: %s, msg: %q)\n",
+			result.Id,
+			codes.Code(result.Status.Code),
+			result.Status.Message)
+	}
+
+	if len(failed) > 0 {
+		return errors.New("failed to delete one or more entries")
```

Comment:
The CLI command should only error if the call as a whole has errored

Reply:
I followed create.go as a guide:
https://github.com/spiffe/spire/blob/5aa2122fbcc892979b15be93a20ebbb6f2c41438/cmd/spire-server/cli/entry/create.go#L284

Maybe I misunderstood the code of entry create, but I think that we are doing the same there.

Thanks for the comments!

Reply:
Ah ok ... consistency is best 😊 thanks for pointing that out